tests/syntax_tests/data/printer/expr/expected/asyncAwait.res.txt

```diff
@@ -29,11 +29,11 @@ user.data = await fetch()
 
 <Navbar promise={await gc()}> {await weirdReactSuspenseApi} </Navbar>
 
-let inBinaryExpression = await x->Js.Promise.resolve + 1
-let inBinaryExpression = await x->Js.Promise.resolve + await y->Js.Promise.resolve
+let inBinaryExpression = (await x->Js.Promise.resolve) + 1
```

Comment:
@nojaf before looking into why printing has changed here, any comments on what the ideal printer should look like, by analogy to the rest of the printer
cc @cknitt

Reply:
here in this file in general

Reply:
Hmm, I'm not entirely sure about this. I believe that no parentheses are needed if the top-level `await` expression is followed by a single expression (top-level from the expression point of view):

```res
await x
```

However, if it is nested within another expression, it makes sense to use parentheses:

```res
let x = foo((await bar), y, z)
let a = (await b) - (await c)
```

I think the changes in this file are reasonable so far. However, I'm no expert on this, so I would like to ask for input from others as well.

Reply:
Looks like the formatting for a simple binary operation is `(await 1) + (await 2)`, so this example looks odd.
I think what's going on before this PR is that by adding await as an attribute, each sub-expressions is processed not as an await expression, but as a `->` expression, as the checks go straight through the ast.
In general `(await e1) + (await e2)` should probably not look inside `e1` and `e2` for deciding how to parenthesise. So it seems the formatting before this PR might not have been intentional.